lang/src/jmh/kotlin/org/partiql/jmh/benchmarks/ParserBenchmark.kt

```diff
@@ -0,0 +1,1368 @@
+                            when albumId = 'default' then 1
+                            when albumId = 'profileimages' then 2
+                            when albumId = 'coverimages' then 3
+                        end as albumType, 
+                        count(imageid) as media_count,  
+                        (NULL) as artist  
+                        FROM userimages
+                        INNER JOIN newsfeedimages on userimages.id = newsfeedimages.imageId
+                        group by newsfeedid 
+                    ) album 
+                ON newsfeed.id = album.newsfeedId 
+            ) 
+            LEFT JOIN 
+            ( 
+            select newsPortalFeedId as feedid,
+            title,description,shortDescription,newsUrl,externalUrlOption, newsPortalCategoryId as pagefeedcategoryid,
+            (15) as defaultCategoryType from newsportalFeedInfo 
+            UNION 
+            select businessPageFeedId as feedid,title,description,shortDescription,newsUrl,externalUrlOption,
+            businessPageCategoryId as pagefeedcategoryid,(25) as defaultCategoryType from businessPageFeedInfo 
+            UNION 
+            select newsfeedId as feedid,(NULL) as title,description,(NULL) as shortDescription,(NULL) as newsUrl,
+            (NULL) as externalUrlOption, categoryMappingId as pagefeedcategoryid,
+            (20) as defaultCategoryType from mediaPageFeedInfo 
+            ) page 
+            ON newsfeed.id = page.feedId WHERE privacy != 10 
+        """.trimIndent()
+
+        val queries = listOf(
+            querySimple, // 0
+            nestedParen, // 1
+            someJoins, // 2
+            severalJoins, // 3
+            someSelect, // 4
+            severalSelect, // 5
+            someProjections, // 6
+            severalProjections, // 7
+            nestedParen, // 8
+            queryFunc, // 9
+            queryFuncInProjection, // 10
+            queryList, // 11
+            query15OrsAndLikes, // 12
+            query30Plus, // 13
+            queryNestedSelect, // 14
+            graphPattern, // 15
+            graphPreFilters, // 16
+            manyJoins, // 17
+            timeZone, // 18
+            caseWhenThen, // 19
+            simpleInsert, // 20
+            exceptUnionIntersectSixty, // 21
+            exec20Expressions, // 22
+            fromLet, // 23
+            groupLimit, // 24
+            pivot, // 25
+            longFromSourceOrderBy, // 26
+            nestedAggregates, // 27
+            complexQuery, // 28
+            complexQuery01, // 29
+            complexQuery02, // 30
+            veryLongQuery, // 31
+            veryLongQuery01, // 32
+        )
+
+        val queriesFailingHalfway = queries.map { query ->
+            val splitQuery = query.split("\\s".toRegex()).toMutableList()
+            val index = (splitQuery.lastIndex * .50).toInt()
+            splitQuery.add(index, ";")
+            splitQuery.joinToString(separator = " ")
+        }
+    }
```

Comment:
Curious on the location of the error chosen here.  It seems like it is always going to be at the end of the query, but it may not necessary be the deepest level of the AST. Will this matter?

Reply:
As long as we're consistent with where we are throwing errors. I did it halfway through the query.